src/R/Components/Impl/InteractiveWorkflow/Commands/SessionInformationCommand.cs

```diff
@@ -35,12 +39,28 @@ public sealed class SessionInformationCommand : IAsyncCommand {
             return CommandResult.Executed;
         }
 
-        private void OnBrokerChanged(object sender, EventArgs e) {
+        private void OnRSessionInitialized(object sender, EventArgs e) {
             if (_interactiveWorkflow.RSession.IsRemote) {
-                PrintBrokerInformationAsync(reportTelemetry: true).DoNotWait();
+                ReplInitComplete().ContinueWith(async (t) => await PrintBrokerInformationAsync(reportTelemetry: true)).DoNotWait();
```

Comment:
> Side note: you can combine non-void returning delegates. But if you invoke the result, you'll only get the last returned value; all others are silently discarded.

I don't remember anything in MSDN that exactly the last one is returned. I think it is considered undetermined. 

As for the event, I think we can raise it when we receive `?>`. What about `RExecutionTracer`? Does it need access to the `RSession` ASAP?